pkg/etcd/etcd.go

```diff
@@ -853,10 +861,152 @@ func (e *ETCD) Snapshot(ctx context.Context, config *config.Control) error {
 		}
 	}
 
+	snapshots, err := e.listSnapshots(ctx, snapshotDir)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Saving current etcd snapshot set to %s ConfigMap", snapshotConfigMapName)
+	return e.storeSnapshotData(ctx, snapshots)
+}
+
+type s3Config struct {
+	Endpoint      string `json:"endpoint,omitempty"`
+	EndpointCA    string `json:"endpointCA,omitempty"`
+	SkipSSLVerify bool   `json:"skipSSLVerify,omitempty"`
+	Bucket        string `json:"bucket,omitempty"`
+	Region        string `json:"region,omitempty"`
+	Folder        string `json:"folder,omitempty"`
+}
+
+// snapshotFile represents a single snapshot and it's
+// metadata.
+type snapshotFile struct {
+	Name string `json:"name"`
+	// Location contains the full path of the snapshot. For
+	// local paths, the location will be prefixed with "file://"
+	// and for S3 will be "s3://".
+	Location  string       `json:"location,omitempty"`
```

Comment:
It sounds like we don't need a generic Location, since:
* For S3 restores we get the bucket/folder info from the S3 field
* For local restores we just need the filename and directory (which could perhaps go in a separate field?)

Reply:
This needs a comment update.